test/unit/mixins/behaviors.spec.js

```diff
@@ -0,0 +1,466 @@
+      expect(BarBehavior.prototype.undelegateEntityEvents).to.have.been.calledOnce;
+    });
+  });
+
+  describe('#_destroyBehaviors', function() {
+    let behaviorsInstance;
+    let FooBehavior;
+    let BarBehavior;
+
+    beforeEach(function() {
+      behaviorsInstance = new Behaviors();
+      FooBehavior = Behavior.extend({});
+      BarBehavior = Behavior.extend({});
+
+      this.sinon.stub(FooBehavior.prototype, 'destroy');
+      this.sinon.stub(BarBehavior.prototype, 'destroy');
+
+      behaviorsInstance.behaviors = {foo: FooBehavior, bar: BarBehavior};
+      behaviorsInstance._initBehaviors();
+    });
+
+    it('should invoke destroy with arguments', function() {
+      let destroyArguments = ['foo', 'bar', 'baz'];
+
+      behaviorsInstance._destroyBehaviors(destroyArguments);
+
+      expect(FooBehavior.prototype.destroy)
+        .to.have.been.calledOnce.and.calledWith('foo', 'bar', 'baz');
+      expect(BarBehavior.prototype.destroy)
+        .to.have.been.calledOnce.and.calledWith('foo', 'bar', 'baz');
+    });
+
+    it('should invoke destroy without arguments', function() {
+      behaviorsInstance._destroyBehaviors([]);
+
+      expect(FooBehavior.prototype.destroy).to.have.been.calledOnce;
+      expect(BarBehavior.prototype.destroy).to.have.been.calledOnce;
+    });
+  });
+
+  describe('#_bindBehaviorUIElements', function() {
+    let behaviorsInstance;
+    let FooBehavior;
+    let BarBehavior;
+
+    beforeEach(function() {
+      behaviorsInstance = new Behaviors();
+      FooBehavior = Behavior.extend({});
+      BarBehavior = Behavior.extend({});
+
+      this.sinon.spy(FooBehavior.prototype, 'bindUIElements');
+      this.sinon.spy(BarBehavior.prototype, 'bindUIElements');
+
+      behaviorsInstance.behaviors = {foo: FooBehavior, bar: BarBehavior};
+      behaviorsInstance._initBehaviors();
+    });
+
+    it('should invoke bindUIElements', function() {
+      behaviorsInstance._bindBehaviorUIElements();
+
+      expect(FooBehavior.prototype.bindUIElements).to.have.been.calledOnce;
+      expect(BarBehavior.prototype.bindUIElements).to.have.been.calledOnce;
+    });
+  });
+
+  describe('#_unbindBehaviorUIElements', function() {
+    let behaviorsInstance;
+    let FooBehavior;
+    let BarBehavior;
+
+    beforeEach(function() {
+      behaviorsInstance = new Behaviors();
+      FooBehavior = Behavior.extend({});
+      BarBehavior = Behavior.extend({});
+
+      this.sinon.spy(FooBehavior.prototype, 'unbindUIElements');
+      this.sinon.spy(BarBehavior.prototype, 'unbindUIElements');
+
+      behaviorsInstance.behaviors = {foo: FooBehavior, bar: BarBehavior};
+      behaviorsInstance._initBehaviors();
+    });
+
+    it('should invoke unbindUIElements', function() {
+      behaviorsInstance._unbindBehaviorUIElements();
+
+      expect(FooBehavior.prototype.unbindUIElements).to.have.been.calledOnce;
+      expect(BarBehavior.prototype.unbindUIElements).to.have.been.calledOnce;
+    });
+  });
+
+  //@todo
```

Comment:
Do you want to get this merged with this one incomplete?  I'm cool with that, just wondering.